stock_incoming_shipment_report/wizard/stock_incoming_shipment_report_wizard.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2019 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+
+
+class StockIncomingShipmentReportWizard(models.TransientModel):
+    _name = 'stock.incoming.shipment.report.wizard'
+
+    date_expected = fields.Datetime('Expected Date')
+
+    @api.multi
+    def generate_stock_incoming_shipment_report_action(self):
+        self.ensure_one()
+        picking_ids = self.env.context.get('picking_ids', [])
+        moves = self.env['stock.picking'].browse(
+            picking_ids).mapped('move_lines').filtered(lambda l: l.date_expected >= self.date_expected)
```

Comment:
@yostashiro I am wondering whether it should be before the date user specify or after.

Reply:
@TimLai125 Should be before.